### PR TITLE
add user property to express.request

### PR DESCRIPTION
**Pull request by JipSterk**
Please fill in this template.

- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/auth0/express-jwt/blob/master/README.md#retrieving-the-decoded-payload

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -77,6 +77,7 @@
 /types/agora-rtc-sdk/                                   @ysclyy @wangjie-agora @disoul
```
Comment (orta) [inline .github/CODEOWNERS:77]: Can you revert this change please?

File: types/express-jwt/index.d.ts [hunk 2]
```diff
@@ -62,3 +62,14 @@ declare namespace jwt {
...
+declare global {
+    namespace Express {
+        // tslint:disable-next-line:no-empty-interface
+        interface User {}
+
+        interface Request {
+            user: User;
+        }
+    }
+}
```
Comment (orta) [inline types/express-jwt/index.d.ts:75]: This seems to match the behavior in https://github.com/auth0/express-jwt/blob/master/README.md#usage 👍🏻

Comment (peterblazejewicz) [reply types/express-jwt/index.d.ts:75]: this also matches the established convention, e.g.:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport/index.d.ts#L19-L24

File: types/express-jwt/index.d.ts [hunk 3]
```diff
@@ -62,3 +62,14 @@ declare namespace jwt {
...
     }
 }
+
+declare global {
+    namespace Express {
+        // tslint:disable-next-line:no-empty-interface
+        interface User {}
+
+        interface Request {
+            user: User;
```
Comment (peterblazejewicz) [inline types/express-jwt/index.d.ts:72]:
```suggestion
            user?: User;
```
can this be optional, please? like:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport/index.d.ts#L24
Not every request/response bears session info of some kind, thx!

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@JipSterk thx!

**Comment by JipSterk** [comment]
Ready to merge

**Comment by AlexRex** [comment]
Maybe joining a bit late this conversation, but wouldn't it make sense to pass the User as generic so we can define the properties instead of having and empty class?

**Comment by peterblazejewicz** [comment]
> Maybe joining a bit late this conversation, but wouldn't it make sense to pass the User as generic so we can define the properties instead of having and empty class?

It could make sense (I personally always fawored module augmentaton, to avoid picking one of the posisble routes).
Mind opening PR with proposed change? There could be many reasons (and gains) behind using non-opaque types with generics, thx!
